pkg/reconciler/delivery/delivery.go

```diff
@@ -92,7 +93,12 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, cfg *v1.Configuration) p
 		return nil
 	}
 
-	// TODO: check NextUpdateTimestamp in the PolicyState for this cfg and enqueue events
+	// check for existing NextUpdateTimestamp to prevent event leaks in case of KCD controller restart, etc.
+	if ps, err := c.fetchPolicyState(cfg); err != nil {
+		return err
+	} else if ps.Status.NextUpdateTimestamp != nil {
+		c.followup(cfg, ps.Status.NextUpdateTimestamp.Time.Sub(c.clock.Now()))
```

Comment:
what if it's _past now_?

Reply:
I modified it so that it ignores when it's past now